jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/KubernetesHelper.java

```diff
@@ -864,8 +856,8 @@ public static File getResourceFragmentFromSource(File resourceDirFinal, List<Str
         return new File[0];
     }
 
-    public static String getFullyQualifiedApiGroupWithKind(CustomResourceDefinitionContext crdContext) {
-        return crdContext.getGroup() + "/" + crdContext.getVersion() + "#" + crdContext.getKind();
+    public static String getFullyQualifiedApiGroupWithKind(String group, String version, String kind) {
+        return group + "/" + version + "#" + kind;
```

Comment:
This seems to be used only for logging purposes. And **always** from data that comes from a GenericKubernetsResource.

You can see the logic repeated both in ApplyService and KubernetesUndeployService.

So at least, this method could be changed to `getFullyQualifiedApiGroupWithKind(GenericKubernetesResource)`, and remove duplications in the calling methods.

Reply:
```suggestion
    public static String getFullyQualifiedApiGroupWithKind(HasMetadata item) {
        return ApiVersionUtil.joinApiGroupAndVersion(ApiVersionUtil.trimGroup(item.getApiVersion(), ApiVersionUtil.trimVersion(item.getApiVersion())) +
        "#" + genericKubernetesResource.getKind());
```